ide/extexecution.base/src/org/netbeans/api/extexecution/base/BaseExecutionService.java

```diff
@@ -140,6 +140,43 @@ public static BaseExecutionService newService(@NonNull Callable<? extends Proces
         return new BaseExecutionService(processCreator, descriptor);
     }
 
+    /**
+     * Infers the output encoding from the relevant system properties, if those should all be <code>null</code>
+     * then this will fallback to <code>Charset.defaultCharset()</code>
+     * 
+     * Since JDK 18 and JEP400 Console.charset() is used for the console's encoding instead of <code>Charset.defaultCharset()</code>. 
+     * Console.charset() is exposed via stdout.encoding/sun.stdout.encoding.
+     * If ran with JDK<=16 these properties should be null and the old default of <code>Charset.defaultCharset()</code> will be used to match pre JEP400 behavior.
```

Comment:
At least `sun.stdout.encoding` may not be `null` on JDK<=16. That predates JEP400.